src/PointwiseFunctions/AnalyticData/GrMhd/CocalInitialData.hpp

```diff
@@ -0,0 +1,216 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+#include <memory>
+#include <mutex>
+
+#include "DataStructures/Tensor/EagerMath/DotProduct.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Evolution/NumericInitialData.hpp"
+#include "IO/External/InterpolateFromCocal.hpp"
+#include "Options/Options.hpp"
+#include "Options/String.hpp"
+#include "PointwiseFunctions/AnalyticData/GrMhd/AnalyticData.hpp"
+#include "PointwiseFunctions/GeneralRelativity/Tags.hpp"
+#include "PointwiseFunctions/Hydro/SpecificEnthalpy.hpp"
+#include "PointwiseFunctions/Hydro/Tags.hpp"
+#include "PointwiseFunctions/InitialDataUtilities/InitialData.hpp"
+#include "Utilities/Serialization/CharmPupable.hpp"
+#include "Utilities/TMPL.hpp"
+#include "Utilities/TaggedTuple.hpp"
+
+/// \cond
+namespace PUP {
+class er;
+}  // namespace PUP
+/// \endcond
+
+namespace grmhd::AnalyticData {
+
+enum class CocalIdType { Co, Ir, Sp };
+
+/*!
+ * \brief Hydro initial data generated by COCAL.
+ *
+ * This class loads numerical data written out by the COCAL initial data solver.
+ *
+ * We choose a constant electron fraction and zero temperature for now because
+ * COCAL doesn't export these quantities. We'll have to improve this later, e.g.
+ * by constructing an EOS consistent with the COCAL data.
+ */
+class CocalInitialData : public evolution::initial_data::InitialData,
+                         public evolution::NumericInitialData,
+                         public AnalyticDataBase {
+ public:
+  struct DataDirectory {
+    using type = std::string;
+    static constexpr Options::String help = {
+        "Path to the directory of data produced by COCAL."};
+  };
+
+  struct ElectronFraction {
+    using type = double;
+    static constexpr Options::String help = {"Constant electron fraction"};
+  };
+
+  struct IdType {
+    using type = CocalIdType;
+    static constexpr Options::String help = {
+        "The ID type of COCAL data (Co, Ir, Sp)"};
```

Comment:
what do these two letter abbreviations mean? Why not use actually telling names?